qiime2/sdk/usage.py

```diff
@@ -240,11 +240,11 @@ def action(self, action: UsageAction, inputs: UsageInputs,
         input_opts = inputs.build_opts(action_signature, self._scope)
         output_opts = outputs.build_opts(action_signature, self._scope)
 
-        derived_outputs = self._action_(action, input_opts, output_opts)
-        self._add_outputs_to_scope(outputs, derived_outputs)
+        computed_outputs = self._action_(action, input_opts, output_opts)
```

Comment:
Does `_action_` still need to return `computed_outputs` here?  It doesn't look like `output_opts` is actually being modified by calls to `_action_ ` in `ArtifactAPIUsage` anymore.  It's not  being modified in `CLIUsage` either.

Reply:
> Does `_action_` still need to return `computed_outputs` here?

Yep! The new schema for `computed_outputs` is `Dict[str, Any]` where the key is name out the registered output on the action, and the value is whatever should be stored in the scope - it can be a computed value, or just some kind of placeholder.

>  It doesn't look like `output_opts` is actually being modified by calls to `_action_ ` in `ArtifactAPIUsage` anymore.

It doesn't necessarily need to be modified, but it does need to be returned. What we are doing in `ArtifactAPIUsage` is returning a `Dict[str, str]` where the keys are the output names and the values are the scope ref names.